src/PrestaShopBundle/Controller/Admin/Improve/Design/CmsPageController.php

```diff
@@ -263,6 +333,12 @@ public function editAction(Request $request, $cmsPageId)
                     ->getUrl($cmsPageId, '{friendly-url}'),
             ]);
             $form->handleRequest($request);
+
+            if ($shopContext == Shop::CONTEXT_SHOP && !empty($addedAssociations)) {
+                foreach ($addedAssociations as $addedShop) {
+                    $this->copyPageContent($cmsPageId, Shop::getContextShopID(), $addedShop);
+                }
```

Comment:
May be move it to a Command Handler would be better ?

Reply:
(and testable)

Reply:
Here again, I'm not ready for this.